pkg/controller/release/observe.go

```diff
@@ -31,6 +40,7 @@ const (
 	errReleaseInfoNilInObservedRelease = "release info is nil in observed helm release"
 	errChartNilInObservedRelease       = "chart field is nil in observed helm release"
 	errChartMetaNilInObservedRelease   = "chart metadata field is nil in observed helm release"
+	errObjectNotPartOfRelease          = "object is not part of release: %v"
```

Comment:
Really happy with where this landed for security reasons and minimizing unexpected behavior. 

I could imagine someone might install a few helm charts and create a final observation, but this puts some rails that no one mis-configures and fetches something from a different helm chart.

The downside of course as discussed is that maybe an operator is installed and simply creates an object that isn't part of the release explicitly. If this ends up being a use case, they could provide an empty secret in the chart, or we could consider in the future an escape config to let them fetch arbitrary resources, but this seems like a great first step.

Reply:
I'm interested in the scenario where I want to observe and provision connection details from a Secret created by an operator. In this case I want to retrieve the kubeconfig from a [vcluster](https://github.com/loft-sh/vcluster/tree/main/charts/k3s) Release. I'm not confident that [the controller](https://github.com/loft-sh/vcluster/blob/3121810c78b63486213d54aa98472ef237d447da/pkg/util/kubeconfig/kubeconfig.go#L27) could handle an existing secret without a patch either. Do I have a way to prevent `object is not part of release` error without forking the charts and creating a dummy secret? It seems like an inelegant and operationally expensive solution.